### DISCUSSION TITLE
there is a requirement,my colleague have vscode

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-glsp/glsp#1293] [Q&A]
there is a intergation requirement,my colleague have vscode extensionA and webviewA,they can communicate with eachother,my glsp project also have a vscode extensionB and webviewB,since my front end colleagues use webviewA,how can he send related action(like create node action) to my vscode extensionB,because my project need my webviewB to trigger related action,but actual webview instance is webview A

**Comment by planger** [comment]
I feel this isn't really related to GLSP but rather related to communicating between VS Code extensions in general.
If one extension needs to invoke some behavior of another one, you could simply trigger a command in extension A that is handled in extension B, which if it is a GLSP extension, send an action to its GLSP server.

If you need a more sophisticated API for extension A provided by extension B, you could expose an API from extension B by returning an object from its `activate` function that is obtained by extension A.
See https://code.visualstudio.com/api/references/vscode-api#extensions
